src/ui/layout/mct-tree.vue

```diff
@@ -822,6 +618,63 @@ export default {
 
                 checkHeights();
             });
+        },
+        getTreeItemByPath(path) {
+            return this.treeItems.find(item => item.navigationPath === path);
+        },
+        getTreeItemIndex(indexItem) {
+            let path = typeof indexItem === 'string' ? indexItem : indexItem.navigationPath;
+
+            return this.treeItems.findIndex(item => item.navigationPath === path);
+        },
+        getChildrenInTreeFor(parent, allDescendants = false) {
+            const parentPath = typeof parent === 'string' ? parent : parent.navigationPath;
+            const parentDepth = parentPath.split('/').length;
+
+            return this.treeItems.filter((childItem) => {
+                const childDepth = childItem.navigationPath.split('/').length;
+                if (!allDescendants && childDepth > parentDepth + 1) {
+                    return false;
+                }
+
+                return childItem.navigationPath !== parentPath
+                && childItem.navigationPath.includes(parentPath);
+            });
+        },
+        isTreeItemOpen(item) {
```

Comment:
i think you can have different checks depending on the item type, rather than modding the item and then running the check

Reply:
Haha, yeah I got a little carried away there, I've refactored to just always pass an item in.